trl/trainer/utils.py

```diff
@@ -265,17 +265,17 @@ def __call__(self, examples: List[Dict[str, Any]]) -> Dict[str, torch.Tensor]:
             assistant_messages = [msg for msg in messages if msg["role"] == "assistant"]
             last_assistant_message = assistant_messages[-1]["content"]
             prompt = formatted_chat.rsplit(last_assistant_message, 1)[0]
-            completion = last_assistant_message
+            completion = last_assistant_message + formatted_chat.rsplit(last_assistant_message, 1)[1]
 
             prompts.append(prompt)
             completions.append(completion)
 
         # Tokenize prompts and completions
         tokenized_prompts = self.tokenizer(
```

Comment:
Let's add a comment here to remind ourselves why we set `add_special_tokens=False`
```suggestion
        # We assume the inputs are already formatted with ChatML, so extra BOS/EOS tokens should not be added.
        tokenized_prompts = self.tokenizer(
```